src/main/java/de/aservo/confapi/commons/model/DirectoryLdapBean.java

```diff
@@ -0,0 +1,80 @@
+package de.aservo.confapi.commons.model;
+
+import de.aservo.confapi.commons.constants.ConfAPI;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * Bean for user directory settings in REST requests.
+ */
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+@XmlRootElement(name = ConfAPI.DIRECTORY_LDAP)
+public class DirectoryLdapBean extends DirectoryBaseBean {
+
+    @XmlElement
+    private ServerSettings serverSettings;
+
+    @XmlElement
+    private Scheme scheme;
+
+    @XmlElement
+    private Permissions permissions;
+
+    @Data
+    @NoArgsConstructor
+    public static class ServerSettings {
+
+        @XmlElement
+        @NotNull
+        @Size(min = 1)
+        private String hostName;
+
+        @XmlElement
+        private int port;
+
+        @XmlElement
+        private boolean useSsl;
+
+        @XmlElement
+        private String username;
+
+        @XmlElement
+        private String password;
+    }
+
+    @Data
+    @NoArgsConstructor
+    public static class Scheme {
+
+        @XmlElement
+        private String baseDn;
+
+        @XmlElement
+        private String userDn;
+
+        @XmlElement
+        private String groupDn;
+    }
+
+    @Data
+    @NoArgsConstructor
+    public static class Permissions {
+
+        @XmlElement
+        private boolean readonly;
+
+        @XmlElement
+        private boolean readonlyForLocalGroups;
+
+        @XmlElement
+        private boolean fullAccess;
+    }
+}
```

Comment:
Ich finde diesen Ansatz hier sehr charmant, hätte aber noch ein paar kleine Änderungsvorschläge. Ich würde diese Untergruppierung dann einfach überall nutzen. Und ich würde dann versuchen die Worte "Settings" und "Configuration" versuchen zu vermeiden, weil es das ja immer ist. Mit meinem "DirectoryConfigurationBean" war ich auch nicht glücklich.

Bei einem Remote Crowd Server sieht das ja so aus:

![image](https://user-images.githubusercontent.com/4580157/86423216-dd1d4c80-bd11-11ea-8118-f085276402a7.png)

Hier würde ich die Gruppen dann so nennen (müssen ja noch nicht einführen, was wir noch nicht nutzen):

- server
- permissions
- advanced

Bei LDAP sieht es so aus:

![image](https://user-images.githubusercontent.com/4580157/86423367-41401080-bd12-11ea-8f0f-4a98f0e13a28.png)

Da hätten wir also:

- server
- schema (mit a am Ende)
- permissions

Und bei dem Internal für Crowd sieht es so aus:

![image](https://user-images.githubusercontent.com/4580157/86423469-7ea49e00-bd12-11ea-8377-67de668b6ec6.png)

Da hätte ich die ganzen Passwort Settings auch gerne in einer extra Gruppe, der Einheitlichkeit halber, aber ich habe noch keinen passenden Namen...

- ???
- permissions

Reply:
guter Vorschlag. Ich packe alles in Gruppen, analog zum UI Aufbau.